metaswitch/common/alarms_parser.py

```diff
@@ -0,0 +1,120 @@
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2015 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+import json
+import argparse
+
+# Dictionary of alarm names -> index. Built up by parsing the JSON file
+alarm_names = {}
+
+# Valid severity levels - this should be kept in sync with the 
+# list in alarmdefinition.h in cpp-common
+valid_severity = {"cleared": "1", 
+                  "indeterminate": "2", 
+                  "critical": "3", 
+                  "major": "4", 
+                  "minor": "5", 
+                  "warning": "6"}
+
+# Valid causes - this should be kept in sync with the
+# list in alarmdefinition.h in cpp-common
+valid_causes = ["software_error", 
+                "database_inconsistency", 
+                "underlying_resource_unavailable"]
+
+# Read in the alarms from a JSON file, and write out the alarm IDs
+# with their index/severity
+def validate_alarms_and_write_constants(json_file, constants_file):
```

Comment:
Worth having a UT for this function (maybe modifying it to return a string rather than write to a file)?